src/bonds.cpp

```diff
@@ -67,15 +67,18 @@ void BondData::shift(int offset) {
 
 bool BondData::hasEnergyFunction() const { return energy != nullptr; }
 
-BondData::~BondData() {}
+BondData::BondData(const decltype(index) index) : index(index) {}
```

Comment:
`decltype` is cryptic and for understanding the signature, I think explicitly using `vector<int>` is better. I'd reserve `decltype()` for more complex types or when using templates. Another alternative is a typedef, but that's also an extra layer.